src/error.rs

```diff
@@ -19,6 +19,8 @@ pub enum Error {
     BadLocation,
     #[error(display = "Failed signature check.")]
     FailedSignature,
+    #[error(display = "Cannot covert a public name into public_key.")]
```

Comment:
Perhaps "Invalid PublicKey bytes," I think this can have a source error from dalek. so s `#[source]` attribute. That will impl a From trait for us